# Competition configuration options
# options:
#   insert_models_over_bins: true     # Whether or not to insert the models that are specified in models_over_bins
#   spawn_extra_models: true          # Whether or not to spawn the models that are specified in models_to_spawn
#   fill_demo_tray: true              # If true, AGV1's tray will have parts from the first order spawned
                                      # directly ontop of it once the competition starts (for testing purposes)
#
# Models to be inserted above the bins
# models_over_bins:
#   bin1:                             # Name of the bin (bin1-bin8, as named in the model)
#     xyz: [1.5, 1.7, 0.75]           # (optional) Origin of the bin, if you want to over-write the default
#     models:                         # List of models to insert
#       model_type_1:                 # Type of model to insert (e.g. gear_part) from local or gazebo database
#         xyz_start: [0.1, 0.2, 0.0]  # Origin of the first model to insert
#         xyz_end: [0.4, 0.6, 0.0]    # Origin of the last model to insert
#         rpy: [0, 0, 0]              # Orientation of all models to insert
#         num_models_x: 4             # How many models to insert along the x dimension
#         num_models_y: 6             # How many models to insert along the y dimension
#
# Models to be spawned in particular reference frames
# models_to_spawn:
#   reference_frame_name:             # Name of the reference frame (e.g. workcell::bin8::link)
#     models:                         # List of models to spawn
#       model_name:                   # Name of the model
#         type: piston_rod_part                 # Type of model
#         pose:
#           xyz: [0.1, 0.1, 0.725]    # Co-ordinates of the model origin in the specified reference frame
#           rpy: [0, 0, '-pi/2']      # Roll, pitch, yaw of the model in the specified reference frame

options:
  insert_models_over_bins: true
  spawn_extra_models: false
  fill_demo_tray: true

models_over_bins:
  bin4:
    models:
      gear_part:
        xyz_start: [0.2, 0.1, 0.0]
        xyz_end: [0.2, 0.5, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 1
        num_models_y: 4

      piston_rod_part:
        xyz_start: [0.5, 0.1, 0.0]
        xyz_end: [0.5, 0.5, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 1
        num_models_y: 6

  bin5:
    models:
      gear_part:
        xyz_start: [0.3, 0.1, 0.0]
        xyz_end: [0.3, 0.5, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 1
        num_models_y: 4

  bin6:
    models:
      gear_part:
        xyz_start: [0.3, 0.1, 0.0]
        xyz_end: [0.3, 0.5, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 1
        num_models_y: 4

  bin3:
    models:
       piston_rod_part:
        xyz_start: [0.1, 0.1, 0.0]
        xyz_end: [0.5, 0.5, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 5
        num_models_y: 5

  bin2:
    models:
       gasket_part:
        xyz_start: [0.1, 0.1, 0.0]
        xyz_end: [0.5, 0.5, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 3
        num_models_y: 3

  bin1:
    models:
       disk_part:
        xyz_start: [0.1, 0.1, 0.0]
        xyz_end: [0.5, 0.5, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 4
        num_models_y: 5

models_to_spawn:
  agv1::kit_tray::tray:
    models:
      gear_part:
        type: gear_part
        pose:
          xyz: [0.1, -0.15, 0]
          rpy: [0, 0, 0]
